### ISSUE TITLE
ng-click doesn't work on md-card > img tag

### ISSUE THREAD

**Issue by angel1st** [corentin-gautier/angular-kaarousel#9]
Hello there,
First, let me tell you, your carousel is actually very good and easy to use.
I have the following issue however and I was wondering if you can help on it - I have `md-card` inside `kaarousel-slide` and this `md-card` has `img` tag with `ng-click` bound. So the sample code is:

    <kaarousel ng-show="sighting.gallery.length"
            layout='row' flex
            ng-cloak
            displayed="options.displayed"
            per-slide="options.perSlide"
            autoplay="options.autoplay"
            animation="options.animation"
            direction="options.direction"
            pause-on-hover="options.pauseOnHover"
            center-active="options.centerActive"
            time-interval="options.timeInterval"
            stop-after-action="options.stopAfterAction"
            hide-nav="options.hideNav"
            hide-pager="options.hidePager"
            nav-on-hover="options.navOnHover"
            pager-on-hover="options.pagerOnHover"
            swipable="options.swipable"
            sync="options.sync"
            loop="options.loop"
            min-width="options.minWidth"
            expand="options.expand"
            always-fill="options.alwaysFill"
            transition-duration="options.transitionDuration"
            before-slide="options.beforeSlide"
            after-slide="options.afterSlide">
            <kaarousel-slide ng-cloak layout='row' flex
                ng-repeat="image in sighting.gallery track by $index">
                <md-card>
                    <img ng-click='showImage($index, $event)' ng-src="{{image.thumbnailUrl}}" class="md-card-image" alt="Sighting Thumbnail"
                    />
                    <md-card-actions layout="row" layout-align="space-between center">
                        <md-button class="md-icon-button" aria-label="Favorite">
                            <md-icon md-svg-icon="lib/materialicons/favorite.svg"></md-icon>
                        </md-button>
                        <md-button class="md-icon-button" aria-label="Share">
                            <md-icon md-svg-icon="lib/materialicons/share.svg"></md-icon>
                        </md-button>
                    </md-card-actions>
                </md-card>
            </kaarousel-slide>
        </kaarousel>

The problem is `showImage()` inside `img` tag is not called. Any help would be appreciated.

**Comment by corentin-gautier** [comment]
@angel1st might be an issue with the swipe option. Have you tried without it ?

**Comment by angel1st** [comment]
@corentin-gautier - nope, it wasn't `swipe` option. What did the trick was simple adding `div` tag as parent if the `img` tag. 
Here is how I fixed it:

    <kaarousel ng-show="sighting.gallery.length"
            ng-cloak
            displayed="options.displayed"
            per-slide="options.perSlide"
            autoplay="options.autoplay"
            animation="options.animation"
            direction="options.direction"
            pause-on-hover="options.pauseOnHover"
            center-active="options.centerActive"
            time-interval="options.timeInterval"
            stop-after-action="options.stopAfterAction"
            hide-nav="options.hideNav"
            hide-pager="options.hidePager"
            nav-on-hover="options.navOnHover"
            pager-on-hover="options.pagerOnHover"
            swipable="options.swipable"
            sync="options.sync"
            loop="options.loop"
            min-width="options.minWidth"
            expand="options.expand"
            always-fill="options.alwaysFill"
            transition-duration="options.transitionDuration"
            before-slide="options.beforeSlide"
            after-slide="options.afterSlide">
            <kaarousel-slide
                ng-repeat="image in sighting.gallery track by $index">
                <md-card>
                    <div ng-click='showImage($index, $event)'>
                        <img ng-src="{{image.thumbnailUrl}}" class="md-card-image image-modal" alt="Sighting Thumbnail"
                     />
                    </div>
                    <md-card-actions layout="row" layout-align="space-between center">
                        <md-button ng-click='likeImage($index, $event)' class="md-icon-button" aria-label="Favorite">
                            <md-icon md-svg-icon="lib/materialicons/favorite.svg"></md-icon>
                        </md-button>
                        <md-button class="md-icon-button" aria-label="Share">
                            <md-icon md-svg-icon="lib/materialicons/share.svg"></md-icon>
                        </md-button>
                    </md-card-actions>
                </md-card>
            </kaarousel-slide>
        </kaarousel>

I am still not sure for the explanation which one causes this - either **angular-kaarousel** control or **md-card** component.

**Comment by corentin-gautier** [comment]
@angel1st wow that is weird ... I don't think it's linked to kaarousel then, because I don't have any special treatment for images

**Comment by angel1st** [comment]
@corentin-gautier - yes, it is, and please note, if I remove **angular-kaarousel** and leave **md-card** only it works just fine, so I guess it is a kind of collision between both components :-)

**Comment by corentin-gautier** [comment]
@angel1st there's a pointer-events: none; on the kaarousel img to avoid the "drag" event from being fired when the user swipes, I can add that only to swipeable carousels (its currently default everytime) but if you need the swipe you will have to override that css rule :)

**Comment by maximelafarie** [comment]
I think I got it! I'll made a PR very soon because I'm currently testing it 😄

**Comment by corentin-gautier** [comment]
@maximelafarie  we just need a 'swipable' class which apply the pointer-events: none :)

**Comment by maximelafarie** [comment]
@corentin-gautier Oh, right! Because I was doing it differently : example here https://goo.gl/hEXix6 . 
You'll see that when you click on any image a console log is triggered by a function in a ng-click and you can swipe the slider too without any ugly drag effect. I was doing it with something like this on the img selector:

```
-moz-user-select: none;
-webkit-user-select: none;
-ms-user-select: none;
user-select: none;
-webkit-user-drag: none;
user-drag: none;
-webkit-touch-callout: none;
```

This way, any event can be triggered on any image in any case and there's no drag effect.
But if you prefer a toggleable class, just tell me my PR is ready! 😉

**Comment by corentin-gautier** [comment]
@maximelafarie yeah that could work but I'm sure you noticed that the click event is fired after the drag end :) which might be considered a bug ...

**Comment by maximelafarie** [comment]
@corentin-gautier Yes you're totally right! :)  So maybe we can do something with the `before slide callback`and the `after slide callback`. In short we want to make the images "ng-clickable" and swipeable but don't want to fire the ng-click event on swipe? ;)

**Comment by angel1st** [comment]
@corentin-gautier @maximelafarie - thanks for the efforts, appreciated.
Let me share with you the workaround I found working for me - I just added a `<div>` tag in between `<md-card>` and its child `<img>` and `ng-click` has been added to the `<div>` tag.
Here is the code snippet:

    <kaarousel ng-show="sighting.gallery.length"
            ng-cloak
            displayed="options.displayed"
            per-slide="options.perSlide"
            autoplay="options.autoplay"
            animation="options.animation"
            direction="options.direction"
            pause-on-hover="options.pauseOnHover"
            center-active="options.centerActive"
            time-interval="options.timeInterval"
            stop-after-action="options.stopAfterAction"
            hide-nav="options.hideNav"
            hide-pager="options.hidePager"
            nav-on-hover="options.navOnHover"
            pager-on-hover="options.pagerOnHover"
            swipable="options.swipable"
            sync="options.sync"
            loop="options.loop"
            min-width="options.minWidth"
            expand="options.expand"
            always-fill="options.alwaysFill"
            transition-duration="options.transitionDuration"
            before-slide="options.beforeSlide"
            after-slide="options.afterSlide">
            <kaarousel-slide
                ng-repeat="image in sighting.gallery track by $index">
                <md-card>
                    
                    <div ng-click='showImage($index, $event)' style="outline: none;">
                        <img ng-src="{{image.thumbnailUrl}}" class="md-card-image image-modal" alt="Sighting Thumbnail"
                     />
                    </div>
                    <md-card-actions layout="row" layout-align="space-between center">
                        <md-button ng-click='likeImage($index, $event)' ng-disabled='image.liked' class="md-icon-button" aria-label="Favorite">
                            <md-icon md-svg-icon="lib/materialicons/favorite.svg"></md-icon>
                        </md-button>
                        <md-badge>{{image.likes | abbreviation}}</md-badge>
                        <md-button class="md-icon-button" aria-label="Share" ng-click='shareImage($index, $event)'>
                            <md-icon md-svg-icon="lib/materialicons/share.svg"></md-icon>
                        </md-button>
                    </md-card-actions>
                </md-card>
            </kaarousel-slide>
        </kaarousel>


Hope this helps you guys, cheers!

Angel

**Comment by corentin-gautier** [comment]
@angel1st yeah but that's a workaround, would be better if it worked without it :)
@maximelafarie Are you still working on this ?

**Comment by angel1st** [comment]
@corentin-gautier - totally agree with you on this!

**Comment by maximelafarie** [comment]
@corentin-gautier Yep I've dev something with an ng-class as you said above and I set this class depending on the `swipe` directive attribute value. I don't know if it's what you really want but it works. Anyway I need to do some more test. 😉
